src/components/PokemonGeneralInfo.js

```diff
@@ -0,0 +1,95 @@
+import React from 'react'
+import {StyleSheet, View} from 'react-native'
+import { useTheme, Text } from 'react-native-paper';
+import PokemonType from './PokemonType';
+
+export default function PokemonGeneralInfo({name, id, weight, height, type1, type2=null}) {
+  const colors = useTheme().colors;
+
+  return ( 
+    <View style={[styles.container, {backgroundColor: colors.card}]}>
+
+      <Text style={styles.name}>
+        {name}
+      </Text>
+
+      <Text style={styles.id}>
+        {"#" + id.padStart(3, "0")}
+      </Text>
+
+      <View style={styles.typesContainer}>
+        <PokemonType type={type1} passedStyle={styles.type}/>
+        {type2 && <PokemonType type={type2} passedStyle={styles.type}/>}
+      </View>
+
```

Comment:
that's cleaner with array approach
```suggestion

      <View style={styles.typesContainer}>
       types.map((type, index) => <PokemonType key={index} type={type} passedStyle={styles.type}/>)
      </View>

```